pie/src/pie_cli/config.py

```
@@ -20,13 +31,48 @@
+                "Downloading Python 3.14 runtime for Python inferlets",
+                total=total if total else None,
+            )
+            chunks = bytearray()
+            for chunk in resp.iter_bytes():
+                chunks.extend(chunk)
+                progress.update(task, advance=len(chunk))
+
+    console.print("[dim]Extracting runtime…[/dim]")
+    decompressed = lzma.decompress(bytes(chunks))
+    with tarfile.open(fileobj=BytesIO(decompressed), mode="r:") as tar:
+        tar.extractall(path=dest_dir)
```

Ruff (0.15.4)

[error] Uses of `tarfile.extractall()`

(S202)